### PR TITLE
docs: end-to-end optimization results for 5 published-FF systems

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

End-to-end optimization for 5 published-FF systems — **with an honest reckoning of what's verified vs what's within measurement noise**. Companion data PR: **ericchansen/q2mm-data#6**. Engine non-determinism root-cause issue: **q2mm#284**.

## Headline results

| System | Reduction | Noise floor | Verdict |
|---|---:|---:|---|
| ch3f | **99.83 %** | 0.00 % | Verified (deterministic) |
| rh-enamide | **44.66 %** | 0.58 % | Verified (77× noise) |
| pd-allyl | 0.13 % | 0.65 % | Within noise — no claim |
| rh-conjugate | 1.4 %* | 2.10 % | Within noise — no claim |
| heck-relay | 3.3 %* | 4.90 % | Within noise — no claim |

(*) reverted pre-final because surrogate-guided step appeared to worsen real OF; both the apparent worsening and the revert are within the per-call noise band.

## The discovery (q2mm#284)

While reviewing the initial PR I found `ObjectiveFunction(x)` is non-deterministic on GPU for the metal-TS systems. Same call, same params, three returns: 6,284,858 / 6,430,076 / 6,492,789 for rh-conjugate.

Per-system noise floor (5 calls, IQR/median): ch3f 0%, rh-enamide 0.58%, pd-allyl 0.65%, rh-conjugate 2.10%, heck-relay 4.90%.

Bisected to two root causes (full diagnosis in q2mm#284):
1. scipy L-BFGS-B Fortran has hidden internal state in the geometry evaluator's inner minimization
2. MM3 energy has non-smooth points where `jax.grad` disagrees with FD by orders of magnitude

## Code changes

- `scripts/regenerate_convergence_results.py`: fixed misleading `final_obj_score` (was JaxLoss surrogate, not real OF). Added `surrogate_improvement_pct` as separate diagnostic field. `improvement_pct` now computed from real OF.
- `q2mm/optimizers/scipy_opt.py`: small comment update on the existing "revert on worsening" safety guard to flag that the comparison is noise-dominated for systems with appreciable non-determinism.

(An earlier revision added a "best-seen real-OF tracker" but was removed in response to PR review — it added expensive real-OF calls per iteration, silently changed `n_evaluations` semantics, and given the noise discovery would only pick the lowest noisy sample with no real signal benefit. The right response to noise is statistical robustness — median-of-N + confidence intervals — which will land separately.)

## Doc changes

- Each system page reports per-call noise floor and labels "Verified" vs "Within noise"
- All 6 copilot-pull-request-reviewer threads addressed (consistency, rounding, nested quotes, the "reverted but values differ" puzzle which is itself a noise symptom, and the best-seen tracker concerns)

## Provenance

- q2mm git_sha: `85dc8c9` (master, post-#281 + #282)
- q2mm-data git_sha: `8b0fcb7`
- jax_devices: `["cuda:0"]` (RTX 5090)

## Closes
- #278 (rh-conjugate non-determinism — partially explained by #284; loader fix in #281 stabilized the ratio range from 0.46/0.96/4602 to 1.02–1.07)

## Out of scope
- Engine fix itself (q2mm#284) — needs jaxopt LBFGS in `JaxEngine.minimize` plus locating + smoothing the MM3 non-smooth term. Separate PR.
- Median-of-N + CI for noise-robust improvement measurement. Separate PR.

### CODE REVIEW

File: docs/systems/small-molecules.md [hunk 1]
```diff
@@ -264,6 +264,34 @@ where multi-start and global search methods show material differences:
...
+| Improvement | **99.83 %** |
+| Iterations / Evaluations | 92 / 2 |
+| Optimizer | L-BFGS-B (scipy) over JaxLoss analytical gradients |
+| Wall time | 3.3 s |
+| Optimized frequency R² | 0.99992 (RMSD 7.81 cm⁻¹) |
+
+This is the largest real-objective improvement of any system in the
+published-FF suite — QFUERZA gives a physically motivated starting
+point with significant remaining headroom, and L-BFGS-B closes
+essentially all of it in 92 iterations.  Raw artifacts at
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/systems/small-molecules.md]: Fixed in d591066. Reworded to "any published-FF benchmark system" — CH3F is the only single-molecule benchmark using qfuerza_fresh, but its still in the same convergence-baseline pipeline as the published TSFFs.

File: docs/systems/rh-conjugate.md [hunk 2]
```diff
@@ -69,24 +69,37 @@ complete failure of cross-engine transfer, not a small miss.
...
-| bond_length | 457 | 0.891 |
-| bond_angle | 926 | 0.454 |
-| eig_diagonal | 1254 | −7.86 |
-
-The dramatic improvement vs the pre-refactor numbers (where
-bond_length R² was −58) is the loss of the QFUERZA overwrite that
-was destroying the published Wahlers fit.  The eigenmatrix R² is
-still negative, reflecting the same MM3* ↔ JAX-engine cross-engine
+| Ratio check | 1.02 (in_band) |
+| Initial ObjectiveFunction score | 6.48 × 10⁶ |
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/systems/rh-conjugate.md:74]: Fixed in d591066. Replaced the hard-coded 1.04 in the success callout with the actual observed range (1.02–1.07) and called out that run-to-run variation reflects the ~2 % GPU noise floor documented in the new caveat below.

File: docs/systems/heck-relay.md [hunk 3]
```diff
@@ -81,9 +81,10 @@ complete failure of cross-engine transfer, not a small miss.
 
 | Metric | Value |
 |--------|:-----:|
-| Ratio check | 1.30 (out_of_band; upper bound 1.15) |
-| Initial ObjectiveFunction score | 3.46 × 10⁶ |
-| Optimization | Skipped at default `ratio_tol=0.15` (borderline — candidate for `ratio_tol=None`) |
+| Ratio check | 1.26 (out_of_band; upper bound 1.15) |
+| Initial ObjectiveFunction score | 3.53 × 10⁶ |
+| Final ObjectiveFunction score | 3.39 × 10⁶ |
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/systems/heck-relay.md]: Fixed in d591066. Excellent catch — you correctly identified that the apparent 3.53M → 3.39M change is impossible after a revert. The reverted parameters are bit-identical to the initial parameters; the 4 % gap in the table is engine non-determinism (q2mm#284), not optimization. Updated the table labels and added an inline note explaining this.

File: docs/systems/heck-relay.md [hunk 4]
```diff
@@ -100,12 +101,30 @@ problem (the published Rosales FF was MM3*-fit and the residual
...
-[`q2mm#276`](https://github.com/ericchansen/q2mm/issues/276) for the
-companion experiment on pd-conjugate.
+### End-to-end optimization with `--ratio-tol none`
+
+Following the [pd-conjugate experiment in #276](https://github.com/ericchansen/q2mm/issues/276),
+heck-relay was also run with `--ratio-tol none` to bypass the gate
+and see whether JaxLoss-guided optimization would still produce a
+useful descent step.  Unlike pd-conjugate (which post-refactor had
+ratio = 0.985 and the bypass was a no-op), heck-relay's ratio is
+genuinely outside band at 1.255 and the surrogate broke down during
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/systems/heck-relay.md]: Fixed in d591066. Standardized on 1.255 in both the table and the prose.

File: q2mm/optimizers/scipy_opt.py [hunk 5]
```diff
@@ -377,6 +377,20 @@ def _run_minimize(
...
+        # an intermediate one in real-OF terms.  Track best-seen real OF
+        # across all accepted iterations so we can recover that point
+        # instead of accepting the surrogate-driven final landing.
+        best_seen = {"score": float(initial_score), "x_active": x0.copy()}
+
+        def _track_best(x_active: np.ndarray) -> None:
+            score = float(objective(np.asarray(x_active, dtype=float)))
+            if score < best_seen["score"]:
+                best_seen["score"] = score
+                best_seen["x_active"] = np.asarray(x_active, dtype=float).copy()
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/scipy_opt.py]: Good catch — agreed and removed in 3573a4b. The cost concern is valid (real-OF call per iteration, expensive for multi-molecule systems) and the silent change to n_evaluations semantics is a real footgun. Adding to that: given the engine non-determinism documented in #284, "best-seen" reduced to picking the lowest noisy sample — no actual signal-vs-noise benefit. The right response to per-call noise is statistical robustness (median-of-N + confidence intervals), not a per-iteration tracker. Will land that separately.

File: docs/systems/heck-relay.md [hunk 6]
```diff
@@ -100,12 +108,40 @@ problem (the published Rosales FF was MM3*-fit and the residual
...
+point (3,392,860 → 3,503,130); `ScipyOptimizer` reverted to the
+initial parameters and the run reports 0.00 % improvement.
+
+!!! warning "Noise floor caveat — the 3.3 % "worsening" is within noise"
+    Repeated GPU `ObjectiveFunction(x0)` calls on heck-relay vary by
+    **~4.9 %** (5-call IQR/median; min 3.36e6, max 3.53e6).  This is
+    the worst noise floor of any system in the published-FF suite.
+    The "3.3 % worse" step that triggered the revert is **inside the
+    noise floor**, so we cannot reliably say the JaxLoss-guided step
+    was actually a bad direction; the comparison itself is noise-limited.
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply docs/systems/heck-relay.md]: Fixed in 3573a4b. Reworded the admonition title to "the 3.3 % apparent worsening is within noise" — no nested quotes.
